src/audio/aria/aria_generic.c

```diff
@@ -40,10 +40,9 @@ inline void aria_algo_calc_gain(struct aria_data *cd, size_t gain_idx,
 		src = audio_stream_wrap(source, src + n);
 		samples -= n;
 	}
-
 	/*zero check for maxis not needed since att is in range <0;3>*/
-	if (max_data > (0x7fffffff >> att))
-		gain = (0x7fffffffULL << 32) / max_data;
+	if (max_data > (0x007fffff >> att))
+		gain = (0x007fffffULL << 32) / max_data;
```

Comment:
<< 32 or << 24? since we are handling with 24LE sample.

Reply:
@btian1 I think the PR seems correct, the resulting is gain is same with max S24 signal value. @fkwasowi please ack